File: app/src/main/java/edu/tjrac/swant/bluetooth/view/ScanInfoFlagsAndServicesFragment.java
package edu.tjrac.swant.bluetooth.view;

import android.annotation.SuppressLint;
import android.bluetooth.le.ScanResult;
import android.os.Bundle;
import android.os.ParcelUuid;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;
import edu.tjrac.swant.unicorn.R;
import edu.tjrac.swant.bluetooth.widget.BlueToothFlagsView;

/**
 * Created by wpc on 2018/2/2 0002.
 */

@SuppressLint("ValidFragment")
public class ScanInfoFlagsAndServicesFragment extends Fragment {

    private final ParcelUuid[] mUuids;
    private final int mAdvertiseFlags;
    @BindView(R.id.ll_services) LinearLayout mLlServices;
    Unbinder unbinder;
    @BindView(R.id.flag_view) BlueToothFlagsView mFlagView;

    public ScanInfoFlagsAndServicesFragment(ScanResult scanResult) {
        mUuids = scanResult.getDevice().getUuids();
        mAdvertiseFlags = scanResult.getScanRecord().getAdvertiseFlags();
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_flags_service, container, false);
        unbinder = ButterKnife.bind(this, view);

        mFlagView.setFlag(mAdvertiseFlags);

        if (mUuids != null && mUuids.length > 0) {
            for (ParcelUuid uuid : mUuids) {
                TextView textView = new TextView(getActivity());
                textView.setText(uuid.toString());
                mLlServices.addView(textView);
            }
        }
        return view;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        unbinder.unbind();
    }
}

File: app/src/main/java/edu/tjrac/swant/bluetooth/view/ScanInfoFlagsAndServicesFragment.kt
package edu.tjrac.swant.bluetooth.view

import android.annotation.SuppressLint
import android.bluetooth.le.ScanResult
import android.os.Bundle
import android.os.ParcelUuid
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import edu.tjrac.swant.unicorn.R
import edu.tjrac.swant.bluetooth.widget.BlueToothFlagsView
import kotlinx.android.synthetic.main.fragment_flags_service.*

/**
 * Created by wpc on 2018/2/2 0002.
 */

@SuppressLint("ValidFragment")
class ScanInfoFlagsAndServicesFragment(scanResult: ScanResult) : Fragment() {

    private val mUuids: Array<ParcelUuid>?
    private val mAdvertiseFlags: Int

    init {
        mUuids = scanResult.device.uuids
        mAdvertiseFlags = scanResult.scanRecord!!.advertiseFlags
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_flags_service, container, false)

        flag_view.setFlag(mAdvertiseFlags)

        if (mUuids != null && mUuids.size > 0) {
            for (uuid in mUuids) {
                val textView = TextView(activity)
                textView.text = uuid.toString()
                ll_services.addView(textView)
            }
        }
        return view
    }

    override fun onDestroyView() {
        super.onDestroyView()
    }
}
